tools/clang/tools/dxc/dxclib/dxc.cpp

```diff
@@ -1011,6 +1011,26 @@ bool GetDLLFileVersionInfo(const char *dllPath, unsigned int *version) {
   return false;
 }
 
+bool GetDLLProductVersionInfo(const char *dllPath, std::string &productVersion) {
+  // This function is used to get product version information from the DLL file.
+  // This information in is not available through a Unix interface.
+#ifdef _WIN32
+  DWORD dwVerHnd = 0;
+  DWORD size = GetFileVersionInfoSize(dllPath, &dwVerHnd);
+  if (size == 0) return false;
+  std::unique_ptr<int[]> VfInfo(new int[size]);
+  if (GetFileVersionInfo(dllPath, NULL, size, VfInfo.get())) {
+    LPVOID pvProductVersion = NULL;
+    unsigned int iProductVersionLen = 0;
+    if (VerQueryValue(VfInfo.get(), "\\StringFileInfo\\040904b0\\ProductVersion", &pvProductVersion, &iProductVersionLen)) {
```

Comment:
Can you add a comment that 040904b0 is a codepage (which one?)

Reply:
Done.